src/Simplex/Chat.hs

```diff
@@ -980,86 +980,64 @@ processChatCommand' vr = \case
+              | localAlias /= "" = localAlias
+              | otherwise = displayName
         CTGroup -> withGroupLock "forwardChatItem, from group" fromChatId $ do
-          gInfo <- withFastStore $ \db -> getGroupInfo db vr user fromChatId
-          (errs, items) <- lift $ partitionEithers <$> withStoreBatch (\db -> map (getGroupCI db) (L.toList itemIds))
-          unless (null errs) $ toView $ CRChatErrors (Just user) errs
-          mapM (ciComposeMsgReq gInfo) items
-          where
-            getGroupCI :: DB.Connection -> ChatItemId -> IO (Either ChatError (CChatItem 'CTGroup))
-            getGroupCI db itemId = runExceptT . withExceptT ChatErrorStore $ getGroupChatItem db user fromChatId itemId
-            ciComposeMsgReq :: GroupInfo -> CChatItem 'CTGroup -> CM ComposeMessageReq
-            ciComposeMsgReq gInfo (CChatItem _ ci) = do
-              (mc, mDir) <- forwardMC ci
-              file <- forwardCryptoFile ci
-              let itemId = chatItemId' ci
-                  ciff = forwardCIFF ci $ Just (CIFFGroup (forwardName gInfo) mDir (Just fromChatId) (Just itemId))
-              pure (ComposedMessage file Nothing mc, ciff)
-              where
-                forwardName :: GroupInfo -> ContactName
-                forwardName GroupInfo {groupProfile = GroupProfile {displayName}} = displayName
+          GroupInfo {groupProfile = GroupProfile {displayName}} <- withFastStore $ \db -> getGroupInfo db vr user fromChatId
+          items <- withFastStore $ \db -> mapM (getGroupChatItem db user fromChatId) itemIds
+          mapM (ciComposeMsgReq $ Just $ CIFFGroup displayName) items
         CTLocal -> do
-          (errs, items) <- lift $ partitionEithers <$> withStoreBatch (\db -> map (getLocalCI db) (L.toList itemIds))
-          unless (null errs) $ toView $ CRChatErrors (Just user) errs
-          mapM ciComposeMsgReq items
-          where
-            getLocalCI :: DB.Connection -> ChatItemId -> IO (Either ChatError (CChatItem 'CTLocal))
-            getLocalCI db itemId = runExceptT . withExceptT ChatErrorStore $ getLocalChatItem db user fromChatId itemId
-            ciComposeMsgReq :: CChatItem 'CTLocal -> CM ComposeMessageReq
-            ciComposeMsgReq (CChatItem _ ci) = do
-              (mc, _) <- forwardMC ci
-              file <- forwardCryptoFile ci
-              let ciff = forwardCIFF ci Nothing
-              pure (ComposedMessage file Nothing mc, ciff)
+          items <- withFastStore $ \db -> mapM (getLocalChatItem db user fromChatId) itemIds
+          mapM (ciComposeMsgReq Nothing) items
         CTContactRequest -> throwChatError $ CECommandError "not supported"
         CTContactConnection -> throwChatError $ CECommandError "not supported"
         where
+          ciComposeMsgReq :: Maybe (MsgDirection -> Maybe Int64 -> Maybe ChatItemId -> CIForwardedFrom) -> CChatItem d -> CM (Either (ForwardFileError, Bool) ComposeMessageReq)
```

Comment:
unclear what's the reason for this refactoring. if there's no reason better to revert it